src/sdc11073/pysoap/msgfactory.py

```diff
@@ -35,12 +36,17 @@ class MessageFactory:
      2) call the serialize method of the CreatedMessage instance to get the xml representation
      """
 
-    def __init__(self, sdc_definitions, logger, validate=True):
+    def __init__(self, sdc_definitions: Type[BaseDefinitions],
+                 additional_schema_specs: Union[List[PrefixNamespace], None],
```

Comment:
`Optional` instead of `Union[..., None]` or preferred solution `list[PrefixNamespace] | None`

Reply:
I do not agree. The last notion is only supported since Python 3.10, but we still support 3.9. The argument is not an optional argument, therefore I prefer the Union notion over the Optional one.